main.go

```diff
@@ -659,7 +739,7 @@ func clientTransport(ctx *cli.Context, enableTLS bool) http.RoundTripper {
 		tr.TLSClientConfig = &tls.Config{
 			RootCAs:            getCertPool(ctx.GlobalString("cacert")),
 			Certificates:       getCertKeyPair(ctx.GlobalString("client-cert"), ctx.GlobalString("client-key")),
-			InsecureSkipVerify: ctx.GlobalBool("insecure"),
+			InsecureSkipVerify: ctx.GlobalBool("insecure") || ctx.GlobalBool("rr-dns-mode"),
```

Comment:
You shouldn't disable TLS verification when using `rr-dns-mode`. As described in https://github.com/minio/sidekick/pull/113#issuecomment-2220353598 (last paragraph), you should create a custom HTTP transport that will connect to the resolved IP address and verify the server certificate that belongs to the hostname that returned the IP address. Disabling TLS verification without the user explicitly asking for it is a no-go.

Reply:
That's why you need a custom HTTP transport. One that dials the IP address and verify the original hostname.

Reply:
This will do the trick:
```go
	client := http.Client{
		Transport: &http.Transport{
			TLSClientConfig: &tls.Config{
				ServerName: "www.example.com",
			},
		},
	}

	resp, err := client.Get("https://93.184.215.14:443")
```
It will connect to IP `93.184.215.14`, but it will validate if the servername is `www.example.com`.